tx.go

```diff
@@ -123,6 +123,15 @@ func (tx *Tx) DeleteBucket(name []byte) error {
 	return tx.root.DeleteBucket(name)
 }
 
+// MoveSubBucket moves a sub-bucket from the source bucket to the destination bucket.
+// Returns an error if
+//  1. the sub-bucket cannot be found in the source bucket;
+//  2. or the key already exists in the destination bucket;
+//  3. the key represents a non-bucket value.
+func (tx *Tx) MoveSubBucket(name []byte, src *Bucket, dst *Bucket) error {
```

Comment:
It seems `tx` is unused here.

Reply:
good catch!

I was thinking to supporting moving top level bucket. In that case, no need to provide the source bucket, the signature & implementation will be below,
```
func (tx *Tx) MoveSubBucket(name []byte, dst *Bucket) error {
    return tx.root.MoveSubBucket(name, dst)
}
```

Reply:
@Elbehery Could you help to update it in this PR?

Reply:
@ahrtr  I am confused now 

how can I specify the root bucket explicitly, it is being init by the tx 

https://github.com/etcd-io/bbolt/blob/4261e220f24e5adf1cdbd877f59de9f9beae528a/tx.go#L54